File: Application.Server/Models/DTOs/GetWorkspaces/WorkspaceGroupDto.cs
using Application.Server.Models.CoworkingDatabase;

namespace Application.Server.Models.DTOs.GetWorkspaces
{
    public class WorkspaceGroupDto
    {
        public WorkspaceType WorkspaceType { get; set; }
        public List<Amenity> Amenities { get; set; } = null!;
        public List<FreeRoomsDto> FreeRooms { get; set; } = null!;
        public List<BookingDto> Bookings { get; set; } = null!;
    }
}

File: application.client/src/DTOs/GetWorkspaces/WorkspaceGroupDto.ts
import { WorkspaceType } from "../../models/WorkspaceType"
import { Amenity } from "../../models/Amenity"
import { BookingDto } from "./BookingDto"
import { FreeRoomsDto } from "./FreeRoomsDto"

export interface WorkspaceGroupDto
{
  workspaceType: WorkspaceType
  amenities: Amenity[]
  freeRooms: FreeRoomsDto[],
  bookings: BookingDto[]
}
